### ISSUE TITLE
Code of Conduct Update

### ISSUE THREAD

**Issue by drsophiemurray** [sunpy/sunpy-project#8]
I think it’s probably time to review/update the current SunPy Code of Conduct (https://sunpy.org/coc). 

Right now it just says to email confidential@sunpy.org to report anything, but it isn’t made clear who is on the CoC enforcement team, and what our handling process is. More transparency might be good so people know what to expect if they report something.

NumFOCUS has offered to help improve our reporting system by providing a form that saves to a Google spreadsheet. Then when NumFOCUS receives notice of a new SunPy CoC report coming in, they can confirm to the person who submitted it that they have received their report and reach out to the SunPy CoC Enforcement Team to let them know that a new report has come in. They recommend against forwarding reports via email and instead to give the appropriate SunPy CoC Enforcement Team members access to the form and the spreadsheet of responses. NumFOCUS role would essentially be to support us on the immediate response side; providing additional eyes to ensure that reports are being seen and confirmed as close as possible to the time when they are submitted. Then they would turn it over to us all to resolve the reports per our process. They are also available to confer with on any given incident, should we want that type of support.

Regarding who to be on the CoC enforcement team, we need at least two people to enforce the CoC to ensure no bias/conflict issues. In other orgs it generally includes a cross-section of people in senior roles. For example, NumFocus CoC Enforcement Team includes the Executive Director, President, and Executive Operations Administrator. 

Regarding what is our handling process, I think we can gain inspiration from NumFocus again! Their CoC is very comprehensive (https://numfocus.org/code-of-conduct). If we use it we just need to give them appropriate credit under Attribution 3.0 Unported (CC BY 3.0). 

Would be really interested to hear everyone’s thoughts on this if you agree with the above, have other ideas, or even just think its fine leave it alone!

**Comment by kakirastern** [comment]
Yes, I agree with the above in principle, especially the need for a transparent processing/handling of cases dealing with violations or breaches of the code of conduct. But in practice, I can also see the need to protect the privacy of the persons involved who will be filing a report/complaint regarding some negative personal experiences or the witnessing of other people being mistreated via open-source related interactions, and also the need to protect the privacy of the individuals being accused. I think in most parts of the modern world especially in the West nowadays, the basic jurisprudent assumption is that one is innocent until proven otherwise, so this consideration should be taken into account as well. So there might be much grey area to be considered and properly addressed if we are to have a more complete "workflow" regarding the CoC issue.

**Comment by drsophiemurray** [comment]
> Yes, I agree with the above in principle, especially the need for a transparent processing/handling of cases dealing with violations or breaches of the code of conduct. But in practice, I can also see the need to protect the privacy of the persons involved who will be filing a report/complaint regarding some negative personal experiences or the witnessing of other people being mistreated via open-source related interactions, and also the need to protect the privacy of the individuals being accused. I think in most parts of the modern world especially in the West nowadays, the basic jurisprudent assumption is that one is innocent until proven otherwise, so this consideration should be taken into account as well. So there might be much grey area to be considered and properly addressed if we are to have a more complete "workflow" regarding the CoC issue.

Yes absolutely, it's something that must be handled sensitively and we should consider how to handle the privacy of those involved. I need to re-read the NumFOCUS CoC but I think it goes into detail about that sort of thing. For example making sure to have different people on the team and different ways to contact them in case an issue is actually about a person on the CoC team itself! We definitely don't want to put anyone off submitting an issue because our processes aren't clear.

**Comment by kakirastern** [comment]
That's great 👍
